daemon/src/file_tracker.rs

```diff
@@ -531,62 +555,83 @@ impl FileTracker {
         is_running: &mut bool,
         rx: &Receiver<FileEvent>,
         scan_stack: &mut Vec<ScanContext>,
-    ) -> Result<Vec<FileTrackerEvent>> {
+    ) -> Vec<FileTrackerEvent> {
         let mut txn = None;
         let mut output_evts = Vec::new();
+        let timeout = Duration::from_millis(100);
+
         while *is_running {
-            let timeout = Duration::from_millis(100);
             select! {
                 recv(rx) -> evt => {
                     if let Ok(evt) = evt {
                         if txn.is_none() {
-                            txn = Some(self.db.rw_txn()?);
-                        }
-                        let txn = txn.as_mut().unwrap();
-                        if let Some(evt) = events::handle_file_event(txn, &self.tables, evt, scan_stack)? {
-                            output_evts.push(evt);
+                            let new_txn = self.db
+                                .rw_txn()
+                                .expect("Failed to renew rw txn");
+
+                            txn = Some(new_txn);
                         }
+
+                        let txn = txn.as_mut().expect("Failed to get txn");
+
+                        events::handle_file_event(txn, &self.tables, evt, scan_stack)
+                            .ok()
+                            .and_then(|evt| evt)
+                            .map(|evt| output_evts.push(evt))
+                            .unwrap_or(());
```

Comment:
Does this drop propagated errors?

Reply:
For now, I've changed it to log. The event handlers were one of the parts I didn't touch for now.